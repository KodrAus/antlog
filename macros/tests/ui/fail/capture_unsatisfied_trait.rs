#![feature(stmt_expr_attributes, proc_macro_hygiene)]

#[macro_use]
extern crate antlog_macros;

// Does not implement `Display`
struct Input;

fn main() {
    call(Input);
}

fn call(input: Input) {

    let kvs: &[(&str, antlog_macros_private::__private::Value)] = &[
        #[display] input,
    ];

    println!("{:?}", kvs);
}
