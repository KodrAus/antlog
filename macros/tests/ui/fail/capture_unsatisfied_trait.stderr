error[E0277]: `Input` doesn't implement `std::fmt::Display`
  --> $DIR/capture_unsatisfied_trait.rs:15:19
   |
15 |         #[display]__log_private_capture!(input),
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Input` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `Input`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `antlog_macros_impl::capture::Capture<antlog_macros_impl::capture::CaptureDisplay>` for `Input`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
